conan/tools/cmake/toolchain/blocks.py

```diff
@@ -656,10 +659,24 @@ class GenericSystemBlock(Block):
         set(CMAKE_SYSTEM_PROCESSOR {{ cmake_system_processor }})
         {% endif %}
 
-        {% if generator_platform %}
+        {% if generator_platform and not winsdk_version %}
         set(CMAKE_GENERATOR_PLATFORM "{{ generator_platform }}" CACHE STRING "" FORCE)
+        {% elif winsdk_version %}
+        if(POLICY CMP0149)
+            cmake_policy(GET CMP0149 _POLICY_WINSDK_VERSION)
+        endif()
+        if(_POLICY_WINSDK_VERSION STREQUAL "NEW")
+            message(STATUS "Conan toolchain: CMAKE_GENERATOR_PLATFORM={{gen_platform_sdk_version}}")
```

Comment:
I think we need to take into account that this is only useful for the "Visual Studio" CMake generators - I'm assuming this block doesn't appear for things like Ninja? (where the SDK version is selected when activating vcvars)

Reply:
Good catch, adding a check that avoids winsdk_verison in toolchain if generator!=Visual